openvdb/points/AttributeSet.h

```diff
@@ -540,6 +558,82 @@ class OPENVDB_API AttributeSet::Descriptor
 #endif
 }; // class Descriptor
 
+
+////////////////////////////////////////
+
+/// @brief A convenience class that collates the Descriptor with metadata that
+/// is only available on each AttributeArray, such as whether the array is
+/// uniform or not. This is useful for querying and during construction of new
+/// AttributeSets.
+/// @note For performance reasons, this class uses copy-on-write for storing
+/// the Descriptor so any modification of the source Descriptor also requires
+/// a deep-copy for the data in this class to remain valid.
+class OPENVDB_API AttributeSet::Info {
+public:
+    struct OPENVDB_API Array {
+        Index stride = 1;
+        bool constantStride = true;
+        bool hidden = false;
+        bool transient = false;
+        bool group = false;
+        bool string = false;
+    }; // struct Array
+
+    /// Default constructor
+    Info() = default;
+
+    /// Construct a new Info class from the given Descriptor.
+    /// @param descriptorPtr the descriptor
+    /// @note Array metadata is default-initialized when using this constructor
+    explicit Info(const AttributeSet::Descriptor::Ptr& descriptorPtr);
+
+    /// Construct a new Info class from the given AttributeSet.
+    /// @param attributeSet the attribute set
+    /// @details This method extracts and stores Array metadata from the
+    /// AttributeArray objects stored within it
+    explicit Info(const AttributeSet& attributeSet);
+
+    /// @brief Return the descriptor
+    Descriptor& descriptor() { return *mDescriptor; }
+    /// @brief Return the descriptor
+    const Descriptor& descriptor() const { return *mDescriptor; }
+    /// @brief Return a shared pointer to the descriptor
+    Descriptor::Ptr descriptorPtr() const { return mDescriptor; }
```

Comment:
Do we really need all 3 of these? The one which returns a ptr provides all access types

Reply:
I don't see a problem with doing this. There's a (small) performance penalty with copying the shared ptr, you use the one that's most appropriate.

There's also precedence for exposing multiple different access methods elsewhere in the codebase (`Grid::baseTreePtr()`, `Grid::baseTree()`, `Grid::constBaseTree()` for example).